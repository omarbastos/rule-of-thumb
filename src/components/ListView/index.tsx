import CardHorizontal from 'components/base/Cards/Horizontal'
import { Poll } from 'utils/interfaces'

interface Props {
  polls: Poll[]
}

const ListView = ({ polls }: Props) => {
  const SKELETONS = 16
  if (polls.length === 0) {
    return (
      <div className="grid grid-cols-1">
        {[...Array(SKELETONS)].map(() => (
          <div className="w-60 h-24 border-2 rounded-md mx-auto my-2">
            <div className="flex animate-pulse flex-row items-center h-full justify-center space-x-5">
              <div className="w-12 bg-gray-300 h-12 rounded-full "></div>
              <div className="flex flex-col space-y-3">
                <div className="w-36 bg-gray-300 h-6 rounded-md "></div>
                <div className="w-24 bg-gray-300 h-6 rounded-md "></div>
              </div>
            </div>
          </div>
        ))}
      </div>
    )
  }
  return (
    <div className="flex flex-col items-center justify-center">
      {polls.map((poll: Poll) => (
        <CardHorizontal key={poll.uid} {...poll} />
      ))}
    </div>
  )
}

export default ListView
