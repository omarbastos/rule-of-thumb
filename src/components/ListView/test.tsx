import { Timestamp } from '@firebase/firestore'
import { render, screen } from '@testing-library/react'

import ListView from '.'

const polls = [
  {
    votes: {
      negative: 36.37,
      positive: 64.63
    },
    picture:
      'https://firebasestorage.googleapis.com/v0/b/rule-of-thumb-oh.appspot.com/o/greta.png?alt=media&token=0475699c-f335-4572-8573-254854cd765a',
    lastUpdated: new Timestamp(1639973375, 379000000),
    name: 'Greta Thumberg',
    description:
      "Thunberg's activism started after convincing her parents to adopt several lifestyle choices to reduce their own carbon footprint.",
    category: 'environment',
    uid: '8x109cEi1pD27j5Hb92Q'
  }
]

describe('<ListView />', () => {
  it('should render ListView as default', () => {
    const { container } = render(<ListView polls={[]} />)

    expect(container).toMatchSnapshot()
  })

  it('should render ListView with data', () => {
    const { container } = render(<ListView polls={polls} />)

    expect(container).toMatchSnapshot()
  })
})
