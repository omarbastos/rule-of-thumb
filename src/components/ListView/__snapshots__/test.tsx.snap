// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListView /> should render ListView as default 1`] = `
<div>
  <div
    class="grid grid-cols-1"
  >
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ListView /> should render ListView with data 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center"
  >
    <div
      class="w-full my-4 h-72 md:h-40 relative "
    >
      <img
        alt="Greta Thumberg"
        class="object-fill  w-full md:w-[18%] h-72 md:h-40"
        src="https://firebasestorage.googleapis.com/v0/b/rule-of-thumb-oh.appspot.com/o/greta.png?alt=media&token=0475699c-f335-4572-8573-254854cd765a"
      />
      <div
        class="absolute left-0 top-0 w-full h-full flex md:bg-gradient-gray"
      >
        <div
          class="w-1/4 h-40"
        />
        <div
          class="p-2 flex w-1/2 space-y-2 flex-col items-start justify-start mt-4 lg:mt-0"
        >
          <h4
            class="text-xl truncate flex-none lg:text-3xl  font-normal text-white"
          >
            Greta Thumberg
          </h4>
          <p
            class="text-sm line-clamp-2 w-full lg:text-base text-white"
          >
            Thunberg's activism started after convincing her parents to adopt several lifestyle choices to reduce their own carbon footprint.
          </p>
        </div>
        <div
          class="flex flex-none w-1/4 mb-4 p-4 justify-end items-center"
        >
          <button
            aria-label="Thumbs Up"
            class="square p-1.5 object-center bg-thumb-up hover:bg-thumb-up-200 lg:h-10 lg:px-4 "
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 16 16"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M3.39556 7.10739C5.33333 4.40864 6.02086 3.63942 5.90612 2.07721C5.79138 0.515007 6.1196 0.085454 6.90037 0.00853218C7.68115 -0.0683897 9.14717 0.334371 9.40453 2.89138C9.51561 3.99767 8.68044 6.17569 8.68044 6.17569L14.5244 6.17871C15.3518 6.17871 16 6.51363 16 7.58103C16 8.58101 14.8311 8.88179 14.8311 8.88179C15.2946 9.07914 15.5889 9.51885 15.579 9.99931C15.5818 10.6772 15.0051 11.2301 14.288 11.237C14.7786 11.4233 15.0587 11.9138 14.9545 12.4038C14.8656 12.991 14.3849 13.4563 13.766 13.5541C14.1713 13.8252 14.3388 14.3165 14.1774 14.7603C13.915 15.3463 13.2225 15.9923 11.6669 15.9923L4.39987 15.9923C3.96738 16.0441 3.54771 15.831 3.35442 15.4616L3.35442 7.43409C3.35141 7.32389 3.36526 7.21385 3.39556 7.10739ZM-1.15248e-06 6.59141L2.84289 6.59141L2.84289 15.9906L-2.79588e-06 15.9906L-1.15248e-06 6.59141Z"
                fill="white"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <button
            aria-label="Thumbs Down"
            class="square p-1.5 object-center bg-thumb-down hover:bg-thumb-down-200 lg:h-10 lg:px-4 mx-2 "
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 16 16"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M12.6044 8.89261C10.6667 11.5914 9.97914 12.3606 10.0939 13.9228C10.2086 15.485 9.8804 15.9145 9.09963 15.9915C8.31885 16.0684 6.85283 15.6656 6.59547 13.1086C6.48439 12.0023 7.31956 9.82432 7.31956 9.82432L1.47561 9.82129C0.64821 9.82129 0 9.48638 0 8.41898C0 7.41899 1.16888 7.11822 1.16888 7.11822C0.705391 6.92086 0.411133 6.48115 0.421017 6.00069C0.418215 5.32278 0.994884 4.76991 1.71195 4.76302C1.22145 4.57668 0.941252 4.08616 1.04546 3.59623C1.1344 3.00902 1.61511 2.54375 2.23399 2.44586C1.82873 2.17481 1.66115 1.68354 1.82258 1.23974C2.08497 0.653748 2.77752 0.00768948 4.33313 0.00768948L11.6001 0.00768948C12.0326 -0.0440559 12.4523 0.168969 12.6456 0.538364L12.6456 8.56591C12.6486 8.67611 12.6347 8.78615 12.6044 8.89261ZM16 9.40859H13.1571L13.1571 0.00941753H16L16 9.40859Z"
                fill="white"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <button
            aria-label="Vote Now"
            class="justify-self-end xl:w-28 bg-black text-base py-1.5 lg:py-2 px-4 whitespace-nowrap text-white border-solid border-2 border-white"
            disabled=""
          >
            Vote Now
          </button>
        </div>
      </div>
      <div
        class="absolute font-bold top-2 right-4 text-sm text-white"
      >
        9 hours ago in environment
      </div>
      <div
        class="absolute top-0 left-0"
      >
        <button
          aria-label="Thumbs Up"
          class="square p-1.5 object-center bg-thumb-up false h-10 px-4"
          disabled=""
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 16 16"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3.39556 7.10739C5.33333 4.40864 6.02086 3.63942 5.90612 2.07721C5.79138 0.515007 6.1196 0.085454 6.90037 0.00853218C7.68115 -0.0683897 9.14717 0.334371 9.40453 2.89138C9.51561 3.99767 8.68044 6.17569 8.68044 6.17569L14.5244 6.17871C15.3518 6.17871 16 6.51363 16 7.58103C16 8.58101 14.8311 8.88179 14.8311 8.88179C15.2946 9.07914 15.5889 9.51885 15.579 9.99931C15.5818 10.6772 15.0051 11.2301 14.288 11.237C14.7786 11.4233 15.0587 11.9138 14.9545 12.4038C14.8656 12.991 14.3849 13.4563 13.766 13.5541C14.1713 13.8252 14.3388 14.3165 14.1774 14.7603C13.915 15.3463 13.2225 15.9923 11.6669 15.9923L4.39987 15.9923C3.96738 16.0441 3.54771 15.831 3.35442 15.4616L3.35442 7.43409C3.35141 7.32389 3.36526 7.21385 3.39556 7.10739ZM-1.15248e-06 6.59141L2.84289 6.59141L2.84289 15.9906L-2.79588e-06 15.9906L-1.15248e-06 6.59141Z"
              fill="white"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        class="absolute bottom-0 overflow-hidden w-full h-10 text-xs flex"
      >
        <div
          aria-label="Greta Thumberg Up Votes"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="63.99"
          class="shadow-none flex justify-start items-center whitespace-nowrap text-white bg-thumb-up"
          role="progressbar"
          style="width: 63.99%; transition: width 2s;"
        >
          <div
            class="flex items-center px-4"
          >
            <svg
              height="22.5px"
              version="1.1"
              viewBox="0 0 16 16"
              width="22.5px"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                id="Mobile"
                stroke="none"
                stroke-width="1"
              >
                <g
                  fill="#FFFFFF"
                  id="Mobile---Home"
                  transform="translate(-56.000000, -334.000000)"
                >
                  <g
                    id="Hero"
                  >
                    <g
                      id="Main-Panel"
                      transform="translate(12.000000, 66.000000)"
                    >
                      <g
                        id="Thumbs-Up"
                        transform="translate(0.000000, 260.000000)"
                      >
                        <g
                          id="thumbs-up"
                          transform="translate(44.000000, 8.000000)"
                        >
                          <path
                            d="M3.39556013,7.10739144 C5.33333333,4.40864038 6.02085655,3.63942071 5.90611697,2.07721333 C5.79137739,0.51500595 6.11959658,0.0854529391 6.90037428,0.00853097137 C7.68115197,-0.0683909963 9.14716722,0.334368969 9.4045313,2.89137618 C9.51561384,3.99766965 8.6804377,6.17568491 8.6804377,6.17568491 L14.524385,6.17870993 C15.3517899,6.17870993 16,6.51362299 16,7.58102333 C16,8.58100891 14.8311191,8.88178245 14.8311191,8.88178245 C15.294609,9.07913941 15.5888675,9.51884816 15.5789835,9.99931171 C15.5817853,10.6772178 15.0051157,11.2300919 14.2880489,11.2369775 C14.7785556,11.4233202 15.0587481,11.9138367 14.9545442,12.4037714 C14.8656004,12.9909767 14.3848923,13.4562488 13.7660067,13.5541438 C14.1712706,13.8251939 14.3388465,14.3164633 14.1774235,14.7602629 C13.915031,15.3462528 13.2224794,15.9923109 11.6668667,15.9923109 L4.39987429,15.9923109 C3.9673802,16.0440571 3.5477081,15.8310308 3.35441845,15.4616357 L3.35441845,7.43409373 C3.35140752,7.32388776 3.36526471,7.21384907 3.39556013,7.10739144 Z M0,6.59140925 L2.8428902,6.59140925 L2.8428902,15.9905823 L0,15.9905823 L0,6.59140925 Z"
                            id="Rectangle_9_copy"
                          />
                        </g>
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </svg>
            <span
              class="ml-2 text-base 2xl:text-2xl"
            >
              63.99
              %
            </span>
          </div>
        </div>
        <div
          aria-label="Greta Thumberg Down Votes"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="36.01"
          class="shadow-none flex justify-end items-center whitespace-nowrap text-white bg-thumb-down/60"
          role="progressbar"
          style="width: 36.01%; transition: width 2s;"
        >
          <div
            class="flex items-center px-4"
          >
            <span
              class="mr-2 text-base 2xl:text-2xl"
            >
              36.01
              %
            </span>
            <svg
              height="22.5px"
              version="1.1"
              viewBox="0 0 16 16"
              width="22.5px"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <title>
                0D3137C2-9DDC-4124-9A31-801057B1DD1F
              </title>
              <desc>
                Created with sketchtool.
              </desc>
              <g
                fill="none"
                fill-rule="evenodd"
                id="Mobile"
                stroke="none"
                stroke-width="1"
              >
                <g
                  fill="#FFFFFF"
                  id="Mobile---Home"
                  transform="translate(-157.000000, -334.000000)"
                >
                  <g
                    id="Hero"
                  >
                    <g
                      id="Main-Panel"
                      transform="translate(12.000000, 66.000000)"
                    >
                      <g
                        id="Thumbs-Down"
                        transform="translate(102.000000, 260.000000)"
                      >
                        <g
                          id="thumbs-up"
                          transform="translate(51.000000, 16.000000) rotate(180.000000) translate(-51.000000, -16.000000) translate(43.000000, 8.000000)"
                        >
                          <path
                            d="M3.39556013,7.10739144 C5.33333333,4.40864038 6.02085655,3.63942071 5.90611697,2.07721333 C5.79137739,0.51500595 6.11959658,0.0854529391 6.90037428,0.00853097137 C7.68115197,-0.0683909963 9.14716722,0.334368969 9.4045313,2.89137618 C9.51561384,3.99766965 8.6804377,6.17568491 8.6804377,6.17568491 L14.524385,6.17870993 C15.3517899,6.17870993 16,6.51362299 16,7.58102333 C16,8.58100891 14.8311191,8.88178245 14.8311191,8.88178245 C15.294609,9.07913941 15.5888675,9.51884816 15.5789835,9.99931171 C15.5817853,10.6772178 15.0051157,11.2300919 14.2880489,11.2369775 C14.7785556,11.4233202 15.0587481,11.9138367 14.9545442,12.4037714 C14.8656004,12.9909767 14.3848923,13.4562488 13.7660067,13.5541438 C14.1712706,13.8251939 14.3388465,14.3164633 14.1774235,14.7602629 C13.915031,15.3462528 13.2224794,15.9923109 11.6668667,15.9923109 L4.39987429,15.9923109 C3.9673802,16.0440571 3.5477081,15.8310308 3.35441845,15.4616357 L3.35441845,7.43409373 C3.35140752,7.32388776 3.36526471,7.21384907 3.39556013,7.10739144 Z M0,6.59140925 L2.8428902,6.59140925 L2.8428902,15.9905823 L0,15.9905823 L0,6.59140925 Z"
                            id="Rectangle_9_copy"
                          />
                        </g>
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
